Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -1310,8 +1310,8 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         }
 
         manager.isOnDemandEnabled = false
-        try await manager.loadFromPreferences()
         try await manager.saveToPreferences()
+        try await manager.loadFromPreferences()
```

Comment:
@diegoreymendez We set `manager.isOnDemandEnabled = false` prior to these 2 calls, so I assume we need to actually `saveToPreferences` to commit it? (The doc for `isEnabled` says ` Note that you will need to re-load the VPN configuration from the preferences in order to see the change in value.` but I'm not sure about `isOnDemandEnabled`.)

Reply:
Also - just wondering: why do we need to save the manager?  This will effectively create a new manager if one doesn't exist.  Sounds to me like if the configuration doesn't exist we should just cancel the tunnel?

Reply:
I'm not sure I follow the problem here. We need to save the manager to commit the change so that Connect on Demand doesn't spin the VPN back up immediately after, and we know the manager already exists because we do `guard let manager = managers.first` right before this.

In this specific example we are then canceling the tunnel and tearing down the extension anyway so any other manager instances won't be affected regardless.

Reply:
@quanganhdo - Going through the code in more detail your change makes sense.  Thanks for taking the time to answer folks!

Reply:
PS: apologies for the mistake!